runtime-sdk/modules/evm/src/evm_backend.rs

```diff
@@ -0,0 +1,224 @@
+//! EVM backend.
+use std::cell::RefCell;
+
+use oasis_runtime_sdk::storage::{self, Store as _};
+
+use crate::types::{H160, H256, U256};
+use evm::backend::{Apply, ApplyBackend, Backend as EVMBackend, Basic, Log};
+
+#[derive(Clone, Eq, PartialEq, cbor::Encode, cbor::Decode, Default)]
+pub struct Vicinity {
+    pub gas_price: U256,
+    pub origin: H160,
+}
+
+#[derive(Clone, Eq, PartialEq, cbor::Encode, cbor::Decode, Default)]
+pub struct Account {
+    pub nonce: U256,
+    pub balance: U256,
+}
+
+pub struct Backend<'c, C: oasis_runtime_sdk::Context> {
+    vicinity: Vicinity,
+    ctx: RefCell<&'c mut C>,
+}
+
+impl<'c, C: oasis_runtime_sdk::Context> Backend<'c, C> {
+    pub fn new(vicinity: Vicinity, ctx: &'c mut C) -> Self {
+        Self {
+            vicinity,
+            ctx: RefCell::new(ctx),
+        }
+    }
+}
+
+/// Prefix for Ethereum accounts in our storage (maps H160 -> Account).
+pub const ACCOUNTS: &[u8] = &[0xea];
+/// Prefix for Ethereum account code in our storage (maps H160 -> Vec<u8>).
+pub const CODES: &[u8] = &[0xec];
+/// Prefix for Ethereum account storage in our storage (maps H160||H256 -> H256).
+pub const STORAGES: &[u8] = &[0xe5];
+
+impl<'c, C: oasis_runtime_sdk::Context> EVMBackend for Backend<'c, C> {
+    fn gas_price(&self) -> primitive_types::U256 {
+        self.vicinity.gas_price.into()
+    }
+    fn origin(&self) -> primitive_types::H160 {
+        self.vicinity.origin.into()
+    }
+    fn block_hash(&self, _number: primitive_types::U256) -> primitive_types::H256 {
+        primitive_types::H256::default()
+    }
+    fn block_number(&self) -> primitive_types::U256 {
+        primitive_types::U256::default()
```

Comment:
We could just use the round number here (e.g., from `self.ctx.runtime_header().round`).